### PR TITLE
[#1793] Initial work making media-view an editor

### PR DESCRIPTION
**Pull request by cadecairos**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/1bb3ed92c434e1a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/1bb3ed92c434e1a/output.txt

Total script time: 0.29 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/4c6b59afa0b2de4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/4c6b59afa0b2de4/output.txt

Total script time: 0.55 mins

Server is running on http://pr715.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/61ac3506a3ab876/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/61ac3506a3ab876/output.txt

Total script time: 0.42 mins

File: src/ui/ui.js [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 define( [ "core/eventmanager", "./toggler",
           "./header", "./unload-dialog",
-          "./tray" ],
+          "./tray", "core/views/media-view" ],
```
Comment (secretrobotron) [inline src/ui/ui.js]: Do you mind renaming all of these files and variables to `-editor` and `___Editor` respectively? It's not a view anymore, right? It's very much an editor.

Comment (cadecairos) [inline src/ui/ui.js]: good call. should I move the file out of src/core/views too?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/49ac4cd843cf4ea/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/49ac4cd843cf4ea/output.txt

Total script time: 0.30 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/22555bfeba55eb2/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/22555bfeba55eb2/output.txt

Total script time: 0.81 mins

Server is running on http://pr715.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/ae8d5c8bab90c88/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/9929643ee27100b/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/ae8d5c8bab90c88/output.txt

Total script time: 2.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/9929643ee27100b/output.txt

Total script time: 2.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/fec2b1bf72d0bf6/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/fec2b1bf72d0bf6/output.txt

Total script time: 0.46 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f2b47edb108ff89/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/a9f7ccaed69e0a5/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f2b47edb108ff89/output.txt

Total script time: 1.75 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a9f7ccaed69e0a5/output.txt

Total script time: 1.14 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

File: src/editor/media-editor.js [hunk 2]
```diff
@@ -2,20 +2,272 @@
...
+
+    function showError( state ) {
+      var inputs = _currentMediaWrapper.querySelectorAll( "input" );
+
+      for ( var i = 0, l = inputs.length; i < l; i++ ) {
+        if ( state ) {
+          inputs[ i ].classList.add( "error" );
+        } else {
+          inputs[ i ].classList.remove( "error" );
+        }
```
Comment (mjschranz) [inline src/editor/media-editor.js:151]: Nit picks, but you could just write something like....

<pre>
inputs[ i ].classList.toggle( "error" );
</pre>


And forego the if statement checking state

File: src/editor/media-editor.js [hunk 3]
```diff
@@ -2,20 +2,272 @@
...
+      }
+      if ( state ) {
+        _mediaErrorMessage.classList.remove( "hidden" );
+      } else {
+        _mediaErrorMessage.classList.add( "hidden" );
+      }
```
Comment (mjschranz) [inline src/editor/media-editor.js:157]: Same as above.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/330e0e3241bfbde/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/330e0e3241bfbde/output.txt

Total script time: 0.74 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/16f1ce34b17df9d/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/af5b39d1cf3689e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/16f1ce34b17df9d/output.txt

Total script time: 1.09 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/af5b39d1cf3689e/output.txt

Total script time: 1.10 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/89fdc1fe5f188ca/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/89fdc1fe5f188ca/output.txt

Total script time: 0.90 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

File: src/core/media.js [hunk 4]
```diff
@@ -7,8 +7,7 @@
             "core/logger",
             "core/eventmanager",
             "core/track",
-            "core/popcorn-wrapper",
-            "core/views/media-view"
+            "core/popcorn-wrapper"
           ],
           function( Logger, EventManagerWrapper, Track, PopcornWrapper, MediaView ){
```
Comment (jbuck) [inline src/core/media.js]: If you're removing the media-view, remove the property for it too

Comment (cadecairos) [inline src/core/media.js]: good catch

**Comment by jbuck** [comment]
Staged on master: https://github.com/mozilla/butter/commit/942740d6154707233aa9f5c14bc723de6f527b96

**Pull request closed without merging**